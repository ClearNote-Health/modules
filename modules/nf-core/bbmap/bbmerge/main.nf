process BBMAP_BBMERGE {
    tag "$meta.id"
    label 'process_single'
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bbmap:39.06--h92535d8_0':
        'biocontainers/bbmap:39.06--h92535d8_0' }"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*_merged.fastq"), emit: merged
    tuple val(meta), path("*_unmerged.fastq"), emit: unmerged
    tuple val(meta), path("*_ihist.txt"), emit: ihist
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    maxmem=\$(echo \"$task.memory\"| sed 's/ GB/g/g')
    bbmerge.sh \\
        -Xmx\$maxmem \\
        in1=${reads[0]} \\
        in2=${reads[1]} \\
        out=${prefix}_merged.fastq \\
        outu1=${prefix}_1_unmerged.fastq \\
        outu2=${prefix}_2_unmerged.fastq \\
        ihist=${prefix}_ihist.txt \\
        $args \\
        &> ${prefix}.bbmerge.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bbmap: \$(bbversion.sh | grep -v "Duplicate cpuset")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_merged.fastq
    touch ${prefix}_1_unmerged.fastq
    touch ${prefix}_2_unmerged.fastq
    touch ${prefix}_ihist.txt
    touch ${prefix}.bbmerge.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bbmap: \$(bbversion.sh | grep -v "Duplicate cpuset")
    END_VERSIONS
    """
}
