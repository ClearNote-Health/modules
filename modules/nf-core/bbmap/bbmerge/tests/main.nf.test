// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test bbmap/bbmerge
nextflow_process {

    name "Test Process BBMAP_BBMERGE"
    script "../main.nf"
    process "BBMAP_BBMERGE"

    tag "modules"
    tag "modules_nfcore"
    tag "bbmap"
    tag "bbmap/bbmerge"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("sarscov2_illumina_paired - fastq_gz") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)]
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2_illumina_paired - fastq_gz - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)]
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
