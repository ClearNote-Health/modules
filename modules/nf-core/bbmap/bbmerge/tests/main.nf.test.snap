{
    "sarscov2_illumina_paired - fastq_gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_merged.fastq.gz:md5,7392aacc8636e65f08b7d7de55647c22"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1_unmerged.fastq.gz:md5,32bdd950ba8393c3069a18aecd81f763",
                            "test_2_unmerged.fastq.gz:md5,4f9d4e1aff34739fb0f4706e51c2bc65"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ihist.txt:md5,710c6563664a8fde42d0d800941add91"
                    ]
                ],
                "3": [
                    "versions.yml:md5,daba7e0cf10dab2c5cdad2d1784a2857"
                ],
                "4": [
                    "test.bbmerge.log:md5,50141eeabef9846959093d56e18e9dee"
                ],
                "ihist": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ihist.txt:md5,710c6563664a8fde42d0d800941add91"
                    ]
                ],
                "log": [
                    "test.bbmerge.log:md5,50141eeabef9846959093d56e18e9dee"
                ],
                "merged": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_merged.fastq.gz:md5,7392aacc8636e65f08b7d7de55647c22"
                    ]
                ],
                "unmerged": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1_unmerged.fastq.gz:md5,32bdd950ba8393c3069a18aecd81f763",
                            "test_2_unmerged.fastq.gz:md5,4f9d4e1aff34739fb0f4706e51c2bc65"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,daba7e0cf10dab2c5cdad2d1784a2857"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-21T13:49:40.456796591"
    },
    "sarscov2_illumina_paired - fastq_gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1_unmerged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2_unmerged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ihist.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,daba7e0cf10dab2c5cdad2d1784a2857"
                ],
                "4": [
                    "test.bbmerge.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "ihist": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_ihist.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    "test.bbmerge.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "merged": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_merged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "unmerged": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1_unmerged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                            "test_2_unmerged.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,daba7e0cf10dab2c5cdad2d1784a2857"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-21T13:49:50.058080222"
    }
}